### DISCUSSION TITLE
stream URL doesn't play on my local player (plays well in web page player)

### DISCUSSION THREAD

**Discussion by andaluspc** [streamlink/streamlink#5789] [Q&A] [answered]
### Description

Hello team.
I use streamlink to extract stream-url of one web page (below are the details), but the stream URL can't be played on vlc or any other local player that I have.
Stream URL has an asterisk (*) in it. Not sure if this is this causes player to see the URL invalid?
I can see same URL structure when I get the stream URL from web browser. However, I have same issue of not being able to play the stream on my local player.
If It try to run the stream-url on browser, Ig et unauthorized (403) http error, which I don't understand because if I open the web page I can stream same URL to my PC.

I'd appreciate anyone that can help me trying to fix this issue

web page: https://www.aloula.sa/en/live/sunnatvsa
sample of stream URL that I can't play: https://live.kwikmotion.com/ksasunnalive/ksasunna.smil/playlist_dvr.m3u8?hdnts=exp=1705844350~acl=/ksasunnalive/ksasunna.smil/*~hmac=317ff8f747f38361f30158453408c52e513f6adab9e55b5034124a20ff3a6917

**Comment by bastimeyer** [comment, marked as answer]
This is a plugin issue. Please open a plugin issue thread instead, because this doesn't belong on the discussion forum.

The plugin is lacking the `Origin` and `Referer` headers, so the HLS multivariant playlist request fails with a 403 status code.

```
$ streamlink --http-header 'Referer=https://www.aloula.sa/' --http-header 'Origin=https://www.aloula.sa' 'https://www.aloula.sa/en/live/sunnatvsa' best
[cli][info] Found matching plugin aloula for URL https://www.aloula.sa/en/live/sunnatvsa
[cli][info] Available streams: 360p (worst), 480p, 1080p (best)
[cli][info] Opening stream: 1080p (hls)
[cli][info] Starting player: /usr/bin/mpv
[cli][info] Player closed
[cli][info] Stream ended
[cli][info] Closing currently open stream...
```

**Reply by andaluspc** [reply]
Thank you Sebastian! I'll manage to raise a plugin issue.
